python/ray/util/queue.py

```diff
@@ -38,9 +41,11 @@ class Queue:
         >>>     assert item == q.get()
     """
 
-    def __init__(self, maxsize: int = 0) -> None:
+    def __init__(self, maxsize: int = 0, actor_options: Optional[Dict] = None) -> None:
+        actor_options = actor_options or {}
         self.maxsize = maxsize
-        self.actor = _QueueActor.remote(self.maxsize)
+        self.actor = ray.remote(_QueueActor).options(**actor_options).remote(
```

Comment:
Why are we removing the decorator from the `_QueueActor` - couldn't we just `_QueueActor.options(**actor_options)` here?

Reply:
This needed if we want to subclass _QueueActor (which I’m doing in the tests in xgboost_ray). If it’s decorated with ray.remote, then we can’t subclass it.